### PR TITLE
toolchain/tasking: add dependencies rules for tasking compiler 

### PR DESCRIPTION
**Pull request by anchao**
## Summary

toolchain/tasking: add dependencies rules for tasking compiler

Tasking compiler uses customized compilation options for generating dependencies files

Signed-off-by: chao an <anchao@lixiang.com>

## Impact

N/A

## Testing

Tasking

### CODE REVIEW

File: tools/Config.mk [hunk 1]
```diff
@@ -636,7 +645,12 @@ endef
 
 export DEFINE_PREFIX ?= $(subst X,,${shell $(DEFINE) "$(CC)" X 2> ${EMPTYFILE}})
 export INCDIR_PREFIX ?= $(subst "X",,${shell $(INCDIR) "$(CC)" X 2> ${EMPTYFILE}})
-export INCSYSDIR_PREFIX ?= $(subst "X",,${shell $(INCDIR) -s "$(CC)" X 2> ${EMPTYFILE}})
+
+ifeq ($(CONFIG_ARCH_TOOLCHAIN_TASKING),y)
+  export INCSYSDIR_PREFIX ?= $(subst "X",,${shell $(INCDIR) "$(CC)" X 2> ${EMPTYFILE}})
+else
+  export INCSYSDIR_PREFIX ?= $(subst "X",,${shell $(INCDIR) -s "$(CC)" X 2> ${EMPTYFILE}})
```
Comment (xiaoxiang781216) [inline tools/Config.mk:652]: should we move the difference to $(INCDIR)?

File: tools/mkdeps.c [hunk 2]
```diff
@@ -115,6 +115,25 @@ static char g_posixpath[MAX_PATH];
 static char g_shquote[MAX_SHQUOTE];
 #endif
 
+static toolchain_t g_toolchain = TOOLCHAIN_GNU;
+
+static const char *g_moptions[TOOLCHAIN_NUM][2] =
```
Comment (xiaoxiang781216) [inline tools/mkdeps.c:120]:
```suggestion
static const char * const g_moptions[TOOLCHAIN_NUM][2] =
```

File: tools/Config.mk (hunk 1 above)
Comment (anchao) [reply tools/Config.mk]: Done

File: tools/mkdeps.c (hunk 2 above)
Comment (anchao) [reply tools/mkdeps.c]: Done

File: tools/Config.mk [hunk 3]
```diff
@@ -268,6 +272,10 @@ else
   INCDIR ?= "$(TOPDIR)/tools/incdir$(HOSTEXEEXT)"
 endif
 
+ifeq ($(CONFIG_ARCH_TOOLCHAIN_TASKING),)
+  INCDIR += -s
```
Comment (xiaoxiang781216) [inline tools/Config.mk:276]: but' it's better to move into incdir.sh like other compiler

Comment (anchao) [reply tools/Config.mk:276]: Incdir has multiple versions, c and shell. so，do we need to apply the similar change for all versions?

Comment (anchao) [reply tools/Config.mk:276]: I think apply changes here would be more reasonable

Comment (xiaoxiang781216) [reply tools/Config.mk:276]: but, all compiler special options is handled in incdir.sh

Comment (anchao) [reply tools/Config.mk]: ok. done

File: tools/mkdeps.c [hunk 4]
```diff
@@ -81,12 +81,12 @@
...
  ****************************************************************************/
 
-enum slashmode_e
+typedef enum
 {
-  MODE_FSLASH  = 0,
-  MODE_BSLASH  = 1,
-  MODE_DBLBACK = 2
-};
+  TOOLCHAIN_GNU      = 0,
```
Comment (xiaoxiang781216) [inline tools/mkdeps.c:86]: should we use the same approach as incdir.c for compiler detecting?

Comment (anchao) [reply tools/mkdeps.c:86]: emm... consider whether it is inappropriate to decide on behavior during compilation? If the toolchain is changed after first configure

File: include/alloca.h [hunk 5]
```diff
@@ -37,6 +37,11 @@
 #  undef  __alloca
 #  define alloca(size)   _alloca(size)
 #  define __alloca(size) _alloca(size)
+#elif defined(__TASKING__)
+#  undef  alloca
+#  undef  __alloca
+#  define alloca(size)   malloc(size)
```
Comment (xiaoxiang781216) [inline include/alloca.h:43]: memory leak?

**Review by xiaoxiang781216 — APPROVED** [review]

File: include/alloca.h (hunk 5 above)
Comment (anchao) [reply include/alloca.h]: Yes, seems still need to improve compatibility on the application side

Comment (anchao) [reply include/alloca.h]: https://github.com/apache/nuttx-apps/pull/2291

Comment (anchao) [reply include/alloca.h]: > can we change sp manually?

how to restore the sp to original position ? setjmp/longjmp?

Comment (xiaoxiang781216) [reply include/alloca.h]: sp should be store in stack in this case

Comment (xiaoxiang781216) [reply include/alloca.h]: if alloca can't implement easily with tasking compiler, it's better to keep alloca is undefined to force the user modify the code to not use it.

**Pull request merged**